sql/core/src/main/scala/org/apache/spark/sql/execution/command/ddl.scala

```diff
@@ -848,4 +851,19 @@ object DDLUtils {
       }
     }
   }
+
+  private[sql] def checkFieldNames(table: CatalogTable): Unit = {
+    val serde = table.storage.serde
+    if (serde == HiveSerDe.sourceToSerDe("orc").get.serde) {
```

Comment:
This way is not right. Let use your previous way with a foreach loop
```
    table.provider.foreach {
      _.toLowerCase(Locale.ROOT) match {
        case "hive" =>
```

Reply:
Yep!